torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -191,6 +191,17 @@ bool needsProfiledOutput(Node* n) {
   }
 }
 
+static void removeProfileCounter(Block* b) {
+  for (auto it = b->nodes().rbegin(); it != b->nodes().rend();) {
+    auto n = *it;
+    if (n->kind() == prim::profile && n->inputs().size() == 0) {
+      it.destroyCurrent();
```

Comment:
If you remove this node the final this will no longer run right ? https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/profiling_record.cpp#L263

Reply:
correct, but we are removing it **after** profiling is done. 

```
  // profile until a graph is ready
  if (!pr_->ready()) {
    return *profiling_plan_;
  }

  auto copy = pr_->graph()->copy();
  ProfilingRecord::removeProfileCounter(copy->block());
  runProfilingOptimizations(copy);
```